common/db.go

```diff
@@ -475,3 +531,113 @@ func buildSelectOrchsQuery(filter *DBOrchFilter) (string, error) {
 	}
 	return query, nil
 }
+
+// FindLatestMiniHeader eturns the MiniHeader with the highest blocknumber in the DB
+func (db *DB) FindLatestMiniHeader() (*blockwatch.MiniHeader, error) {
+	row := db.findLatestMiniHeader.QueryRow()
+	var (
+		number  int64
+		parent  string
+		hash    string
+		logsEnc string
+	)
+	if err := row.Scan(&number, &parent, &hash, &logsEnc); err != nil {
+		return nil, fmt.Errorf("could not retrieve latest header: %v", err)
+	}
+
+	logs, err := decodeLogsJSON([]byte(logsEnc))
+	if err != nil {
+		return nil, err
+	}
+	return &blockwatch.MiniHeader{
+		Number: big.NewInt(number),
+		Parent: ethcommon.HexToHash(parent),
+		Hash:   ethcommon.HexToHash(hash),
+		Logs:   logs,
+	}, nil
+}
+
+// FindAllMiniHeadersSortedByNumber returns all MiniHeaders in the DB sorting in descending order by block number
+func (db *DB) FindAllMiniHeadersSortedByNumber() ([]*blockwatch.MiniHeader, error) {
+	var headers []*blockwatch.MiniHeader
+	rows, err := db.findAllMiniHeadersSortedByNumber.Query()
+	if err != nil {
+		return nil, err
+	}
+	for rows.Next() {
+		var (
+			number  int64
+			parent  string
+			hash    string
+			logsEnc string
+		)
+		if err := rows.Scan(&number, &parent, &hash); err != nil {
```

Comment:
Need to pass `&logsEnc` into `rows.Scan()` or else it will always be the empty string

Reply:
Fixed in ced063e